auto_tuning/proxy/src/proxy_seissol_allocator.hpp

```diff
@@ -248,13 +250,37 @@ void initDataStructuresOnDevice() {
   m_ltsTree->allocateScratchPads();
 
 
-  seissol::initializers::recording::CompositeRecorder recorder;
+  seissol::initializers::recording::CompositeRecorder<seissol::initializers::LTS> recorder;
   recorder.addRecorder(new seissol::initializers::recording::LocalIntegrationRecorder);
   recorder.addRecorder(new seissol::initializers::recording::NeighIntegrationRecorder);
 
 #ifdef USE_PLASTICITY
   recorder.addRecorder(new seissol::initializers::recording::PlasticityRecorder);
 #endif
   recorder.record(m_lts, layer);
+  if (enableDynamicRupture) {
+    auto &drLayer = m_dynRupTree->child(0).child<Interior>();
+    const auto drLayerSize = drLayer.getNumberOfCells();
+    constexpr size_t QInterpolatedSize = CONVERGENCE_ORDER * tensor::QInterpolated::size() * sizeof(real);
+    constexpr size_t imposedStateSize = tensor::QInterpolated::size() * sizeof(real);
+    constexpr size_t idofsSize = tensor::Q::size() * sizeof(real);
+
+    drLayer.setScratchpadSize(m_dynRup.QInterpolatedPlusOnDevice, QInterpolatedSize * drLayerSize);
+    drLayer.setScratchpadSize(m_dynRup.QInterpolatedMinusOnDevice, QInterpolatedSize * drLayerSize);
+    drLayer.setScratchpadSize(m_dynRup.idofsPlusOnDevice, idofsSize * drLayerSize);
+    drLayer.setScratchpadSize(m_dynRup.idofsMinusOnDevice, idofsSize * drLayerSize);
+
+    constexpr auto UpperStageFactor = dr::pipeline::DrPipeline::TailSize * dr::pipeline::DrPipeline::DefaultBatchSize;
```

Comment:
Why is this in upper case?

Reply:
`UpperStageFactor` is in upper case because it is a `constexpr`. I thought we have a convention that all constants should start with upper case.

DrPipeline is derived from GenericPipeline pipeline and it has only 3 stages. 

I guess the following figure make the terminology clear:

![pipeline](https://user-images.githubusercontent.com/19637079/114707705-d43eba00-9d2a-11eb-94a9-ea6fb0f05ada.png)

Reply:
Hm, I would introduce upper case for constants (except maybe static constants) just because I try to declare everything const that can be const.